### PR TITLE
Add readme for debugging api code using k8s resources

### PR DESCRIPTION
**Pull request by thearifismail**
# Overview

This PR adds a reame.md document for how to set up local development environment for debugging local code using resources deployed in a K8S resources.  This is applicable to internal development for now.  

The reason for writing this document is that developers can create hosts using `make run_inv_mq_service_test_producer` task in the make file and then get the newly created host using `curl`.  Due to the removal searches using DB, it was not possible.

## PR Checklist

- [ ] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: docs/connect_to_k8s_env.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# Deploy Host Inventory to a Kubernetes Namespace in Ephemeral Cluster
+
+Debugging the Host Inventory API code needs database, kakfa broker, Kafka Zookeeper, Debezium Connector, and xjoin-search.  Following is a list high level steps necessary to debug local API code using an Ephemeral namespace.  The steps are the same for resources runnign in minikube.
```
Comment (strider) [inline docs/connect_to_k8s_env.md:3]: Following is a list **of** high level steps

File: docs/connect_to_k8s_env.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
+# Deploy Host Inventory to a Kubernetes Namespace in Ephemeral Cluster
+
+Debugging the Host Inventory API code needs database, kakfa broker, Kafka Zookeeper, Debezium Connector, and xjoin-search.  Following is a list high level steps necessary to debug local API code using an Ephemeral namespace.  The steps are the same for resources runnign in minikube.  
+
+1. Login to Epheremal cluster.
```
Comment (strider) [inline docs/connect_to_k8s_env.md:5]: `s/Epheremal/Ephemeral/`

File: docs/connect_to_k8s_env.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    Though API GET is used, the host has been provided by `xjoin-search` which gets it from `elasticsearch` index.
+11. To get hosts directly from the Elasticsearch index, run:
+    ```bash
+        curl --location 'http://localhost:9200/xjoin.inventory.hosts/_search' \
+             --header 'Authorization: Basic <elastic-user value from get_credentials>='
+    ```
+    Note: index name, "xjoin.inventory.hosts" may be different in your case.
+
+# Setup Development environment for Debugging API Code
+This section describes how to launch debugger using `vscode` to execute the local API server which in turn uses resources deployed in the ephemeral namespace.
```
Comment (strider) [inline docs/connect_to_k8s_env.md:58]: `s/vscode/VS Code/`

File: docs/connect_to_k8s_env.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+1. Load the `host-inventory` project to vscode
```
Comment (strider) [inline docs/connect_to_k8s_env.md:59]: `s/vscode/VS Code/`

File: docs/connect_to_k8s_env.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+2. Add a launch configuration to run the api-server (run.py) with the following environment variables:
```
Comment (strider) [inline docs/connect_to_k8s_env.md:60]: Use backticks for `run.py` Python script.

File: docs/connect_to_k8s_env.md (hunk 1 above)
Comment (strider) [inline docs/connect_to_k8s_env.md:3]: Extra space before the second sentences starting by `Following`

**Review by strider — CHANGES_REQUESTED** [review]
Just a few minors issues. Otherwise, it looks good to me.

**Review by jpramos123 — COMMENTED** [review]
some suggestions

tested and this works just fine

**Review by kruai — CHANGES_REQUESTED** [review]
Had a couple comments in addition to the existing suggestions. Thanks for writing this up!

**Review by CodeHeeler — APPROVED** [review]
LGTM

**Review by kruai — APPROVED** [review]

**Review by strider — APPROVED** [review]
LGTM

File: README.md [hunk 6]
```diff
@@ -372,3 +372,7 @@ make update-schema
 
 This will pull the latest version of the System Profile schema from inventory-schemas and update files as necessary.
 Open a PR with these changes, and it will be reviewed and merged as per [the standard process](#release-process).
+
+## Debug API Code using Ephemeral Cluster
+
+Instructions for setting up development environment for stepping through API code is provided [here](docs/connect_to_k8s_env.md)
```
Comment (strider) [inline README.md:378]: Run `pre-commit run --all-files` to fix the end of files hook!

image

**Review by strider — CHANGES_REQUESTED** [review]

**Review by strider — APPROVED** [review]

**Pull request merged**